components/director/internal/open_resource_discovery/client_test.go

```diff
@@ -26,39 +29,120 @@ func NewTestClient(fn RoundTripFunc) *http.Client {
 	}
 }
 
+var successfulRoundTripFunc = func(t *testing.T) func(req *http.Request) *http.Response {
+	return func(req *http.Request) *http.Response {
+		var data []byte
+		var err error
+		var reqBody []byte
+		if req.Body != nil {
+			reqBody, err = ioutil.ReadAll(req.Body)
+			require.NoError(t, err)
+		}
+		statusCode := http.StatusOK
+		if strings.Contains(req.URL.String(), ord.WellKnownEndpoint) {
+			data, err = json.Marshal(fixWellKnownConfig())
+			require.NoError(t, err)
+		} else if strings.Contains(req.URL.String(), ordDocURI) {
+			data, err = json.Marshal(fixORDDocument())
+			require.NoError(t, err)
+		} else if strings.Contains(string(reqBody), "client_secret") {
+			statusCode = http.StatusOK
+			data, err = json.Marshal(struct {
+				AccessToken string `json:"access_token"`
+			}{
+				AccessToken: "test-tkn",
+			})
+			require.NoError(t, err)
+		} else if strings.Contains(string(reqBody), "grant_type=client_credentials") {
+			statusCode = http.StatusOK
```

Comment:
I think this case will match the if above and never reach this one?

Reply:
The oauth lib sends two requests, the first one is matched by the above case, and the second one by this here